visualization/app/codeCharta/state/angular-redux/createSelector.ts

```diff
@@ -15,6 +15,10 @@ export function createSelector<S1, S2, S3, S4, P, State = CcState>(
 	selectors: [Selector<S1, State>, Selector<S2, State>?, Selector<S3, State>?, Selector<S4, State>?],
 	projector: (s1: S1, s2?: S2, s3?: S3, s4?: S4) => P
 ): (state: State) => P
+export function createSelector<S1, S2, S3, S4, S5, P, State = CcState>(
+	selectors: [Selector<S1, State>, Selector<S2, State>?, Selector<S3, State>?, Selector<S4, State>?, Selector<S5, State>?],
+	projector: (s1: S1, s2?: S2, s3?: S3, s4?: S4, s5?: S5) => P
+): (state: State) => P
```

Comment:
I would love to find a solution to prevent the [death by a thousand overloads](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html) but so far I can't think of how to implement that.

Reply:
Me two. I took a quick look into how NgRx does it. Unfortunately that is how they do it as well.